bindings/experimental/distrdf/python/DistRDF/PythonMergeables.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import annotations
+
+from typing import Union, List, TYPE_CHECKING
+
+import ROOT
+import os
+from ROOT._pythonization._rdataframe import AsNumpyResult
+
+if TYPE_CHECKING:
+    from DistRDF.Backends.Base import BaseBackend
+
+
+class SnapshotResult(object):
+    """
+    Encapsulate information coming from a Snapshot operation and know how to
+    merge it with other objects of this type.
+    """
+
+    MERGE_OUTPUT = True
+
+    def __init__(self, treename: str, filenames: List[str], resultptr: ROOT.RDF.RResultPtr = None) -> None:
+        self.treename = treename
+        self.filenames = filenames
+        # Transient attribute, it will be discarded before the end of the mapper
+        # function (in `Utils.get_mergeablevalue`) so that we don't incur in
+        # serialization of the RResultPtr
+        self._resultptr = resultptr
+
+    def Merge(self, other: SnapshotResult) -> None:
+        """
+        When calling Snapshot on a distributed worker, a list with the path to
+        the snapshotted file on the worker is stored. This function extends the
+        list of the current object with the elements from the list of the other
+        object.
+        """
+        self.filenames.extend(other.filenames)
+
+    def GetValue(self, backend: BaseBackend):
+        """
+        With local RDataFrame, Snapshot returns another RDataFrame object that
+        can be used to continue the application. The equivalent in the
+        distributed scenario is to create a distributed RDataFrame.
+
+        This is done by constructing a TChain with the name and the list of
+        paths stored in this object. The chain is then passed to the
+        `make_dataframe` function that changes depending on the backend.
+
+        For example, if the original RDataFrame that triggered the distributed
+        computation was created via a Spark backend, then this function will
+        return another distributed RDataFrame build from a Spark backend
+        instance. And so on for all other DistRDF backends.
+
+        If MERGE_OUTPUT is True, it will merge all partial files into a single output file
+        """
+        snapshot_chain = ROOT.TChain(self.treename)
+        # Add partial snapshot files to the chain
+        for filename in self.filenames:
+            snapshot_chain.Add(filename)
+
+        if SnapshotResult.MERGE_OUTPUT and len(self.filenames) > 1:
+            output_path = self._get_base_filename()
+            merged_file_path = self._merge_snapshot_files(output_path)
+            if os.path.exists(merged_file_path):
+                self._cleanup_partial_files()
+            merged_chain = ROOT.TChain(self.treename)
+            merged_chain.Add(merged_file_path)
+            self.filenames = [merged_file_path]
+            return backend.make_dataframe(merged_chain)
```

Comment:
This logic is not enough, it does not take into account tree reduction and it forces the creation of the output TChain at creation time of the SnapshotResult object. Furthermore, it does not even take into account the possibility of RNTuple Snapshot